### ISSUE TITLE
Route not marked to be linked to a topo

### ISSUE THREAD

**Issue by birgander2** [theCrag/website#2110]
When a route is on a topo, which is not placed in the same node, the route number is not marked in black as usual. Example:
Node: http://www.thecrag.com/climbing/france/fontainebleau/area/399797829
Route: http://www.thecrag.com/climbing/france/fontainebleau/route/524153841

Additionally (probably related): When clicking "edit" on the topo shown in the route details, a page without function opens.

**Comment by brendanheywood** [comment]
The bug with the editing link I have hot fixed.

If we showed the numbers from routes in another areas you can end up with the same number duplicated which is confusing. There probably needs to be more discussion here around how to best model the circuits

**Comment by birgander2** [comment]
I mean just to paint the number box left of the route name in black to indicate that a topo will be shown up when I click on it. That would be absolutely sufficient.

**Comment by brendanheywood** [comment]
Yeah sorry I misunderstood. To me the black label means it is shown in a topo on *this page* and if we swapped it then there are other use cases that would also get changed which I'm not sure I'd want. 

If you need to drill into a route level page to see the topo then I don't think that's very usable, you want the topos interleaved with the routes. Still thinking about to best do this with the circuit concept, I'm wondering wether we could experiment with the gym color overrides in this context. @scd?

**Comment by scd** [comment]
In a gym the color override is used for the grade, but we want color override to refer to route number in Font. Maybe an extra field?

Maybe the indoor vs outdoor usage is sufficient to trigger the different use cases.

Have we ruled out showing the topos in the circuit page. I know this means a lot of topos, when really you are probably more interested in a concise list. 

I also note that in the same topo on the following page is showing a route number from another page, which is potentially confusing:

http://www.thecrag.com/climbing/france/fontainebleau/area/83166210

A route number color will potentially sort this, at least for the Font use case. My thoughts are as follows:
 - [ ] we should not use grade colors on route numbers in topos anywhere.
 - [ ] if a topo has a route number which is not from the current page then it should have some sort of indicator (eg .3 - ie dot before the number).
 - [ ] add a route number color config variable
 - [ ] if there is a route number color then that will be used in the topo and the route list.
 - [ ] always highlight a route is in a topo even if the topo is not on a page.

I think we can do the last of these independently of all others (maybe use a lighter grey).

**Comment by brendanheywood** [comment]
Before we do anything we should really resolve 2 things: what is the perfect way to model circuits in font (and elsewhere?) and what are the perfect way(s) to show them?

Simply adding a color field is one possibility but circuits are also ordered. If we put all routes together into one area and then assign each route to a circuit then the numbers are off, or we nee to account for this. If we leave them as different areas they work ok as circuits but mess up lots of other things like the maps. And unlike gyms where colors can be arbitrary and open ended, here colours are predefined and very limited and are almost a gradeband but not quite as they overlap a lot.

If we change the business rules for when topos are show then that opens up a can of worms which we've been through before multiple times and put a lot of thought into, showing these topos at a node which is note the topo parent will cause regressions elsewhere. 

And what if you don't want to do a circuit but just want to look at a rock and figure out what is what? The circuit structure makes this fairly difficult.

Maybe the end solution will looks something like at the crag level you define the list of available circuits, which assumes a route can only be in one circuit (is this true?) and then you assign all routes to one. In list view you can either see all routes more or less like you would at any other place in the world (except it would also show the right numbers and colors for the circuits) or you can filter it to a specific circuit. This probably best models what the font guides do, and gives the most flexibility to use it in different ways.

**Comment by scd** [comment]
Are the routes in a circuit physically numbered? 

I note that in Font, each area can have multiple circuits with 50+ routes in each circuit. That is a lot of routes for a single area if we collapse.

**Comment by brendanheywood** [comment]
Sometimes, but not consistently, and I'd say the practice is fading.

And yeah some areas are massive like http://www.thecrag.com/climbing/france/fontainebleau/area/14018443 so collapsing isn't perfect either. 

Also whether we are coming at this just from a ticking-to-support-the-guidebooks point of view, or a guide in it's own right would also affect the solution.

**Comment by scd** [comment]
guidebook.

**Comment by birgander2** [comment]
I see your points, but overall it is maybe not perfect as it is, but quite ok. I just thought that it might be helpful to recognise already in the route/circuit list that the route is drawn on a topo, i.e. that there is some additional beta hidden. Indeed I think to keep the circuits as a pure list without interleaved topos is better. For ticking you go into the circuits list, for 'guidbook mode' you stay one level below. For me that's absolutly fine. 

Most circuits in Bleau are physically numbered (and there are various additional signs, direction, etc.) However, depending when they have been repainted the last time, colour has sometimes faded. To my knowledge, each route is only in one circuit. Concerning collapsing: Don't forget the 100+ off-circuit problems you have additionally to the circuit ones.

**Comment by scd** [comment]
@birgander2 we are not on the ground there so your input is very appreciated on this. 

@brendanheywood can we add an indicator to a route that route is linked to a topo when the topo is not displayed on the page. I guess this would have to come from the template generation rather than topo code itself. Could we just use class="hidden-topo" or something similar. Will this have impacts on the topo rules or code? 

I will probably have to add some template data

**Comment by brendanheywood** [comment]
We could always just make a third label color, maybe a darkish gray which says, I'm in a topo, but it's not here. And yes this would be new template data but would not be topo code at all, it's just the route linkage. What about the case where a route is linked but not drawn?

**Comment by scd** [comment]
+1

Are there any use cases when we should have a linked topo, but not drawn.

**Comment by birgander2** [comment]
The only use case I can think of: reminder to finally draw the line (light gray button)
